GDJS/Runtime/variable.js

```diff
@@ -4,21 +4,36 @@
  * This project is released under the MIT License.
  */
 
+/**
+ * @typedef variableData Data representation of a GDevelop variable
+ * @property {string} [name] The name of the variable. Used if a child variable.
```

Comment:
You don't need the [ ] ? There are the other 3 lines

Reply:
The [var] is for declaring a variable optional/that doesn't make it crash if it is not defined in JSDoc

Reply:
Ah nice I didn't know! name is mandatory but let's keep that for the two next properties then :)

Reply:
With non mandatory I mean no code breaking. As there is a check for name definition and it automatically gives it a default value if it isn't, the code won't break if you don't give the name of the object